### DISCUSSION TITLE
Moving from drake to targets: package management and documentation recommendations

### DISCUSSION THREAD

**Discussion by jennysjaarda** [ropensci/targets#487] [Help] [answered]
Hello! I'm very excited to finally be migrating from `drake` to `targets` and had a few questions: 

(1) In my drake projects, I had been using [`workflowr`](https://jdblischak.github.io/workflowr/articles/wflow-01-getting-started.html) to document my project and share with collaborators. It worked well, but I essentially updated and ran Rmd files completely independent of the drake pipelines, using `loadd` and `readd` to show results and having `drake` to the bulk of the analysis (i.e. no major/expensive analyses were happening in the Rmd file). With `targets`, I see there is the `tar_render` which is possibly a better option. Do you see these two packages as compatible? Also how do you recommend integrating the new Target Markdown features into the workflow with multiple report files? I'm struggling to see what exactly the new feature adds over just creating Rmarkdown files and using `tar_render`. Any suggestions and advice would be welcome!

(2) I previously was using packrat to manage my project packages. I've seen you also discuss `renv`. Would this be a better option for use with `targets`? 

Thanks in advance for your suggestions. 

Jenny

**Comment by wlandau** [comment, marked as answer]
`tarchetypes::tar_render()` is like `drake::knitr_in()`. It allows you to write reports as small pieces of a pipeline that automatically re-render when the upstream targets change. These reports are usually small one-page documents mainly for summarizing upstream computations. Target Markdown, on the other hand, tries to be more comprehensive. I have not personally tried it yet with a multi-page project with `bookdown` or `workflowr`, but in theory it should be easier to work into longer book-like workflows. For example, you could distribute your target definitions over multiple chapters and then have another chapter or chapters later on to run the pipeline and load the results. Alternatively,  you could have multiple reports that just write the scripts to define the pipeline, and you could call `tar_make()` etc. outside the literate programming documents. The specific usage patterns are still in flux, but I think there is room to find out what works for you.

I would definitely recommend `renv` over `packrat`. Both are compatible with `targets`,  but `renv` is much more efficient. You might have a look at `targets::tar_renv()`, which generates a script file to make hidden package dependencies more explicit for a project.
